src/gennylib/src/context.cpp

```diff
@@ -205,4 +214,55 @@ bool PhaseContext::isNop() const {
     auto& nop = (*this)["Nop"];
     return nop.maybe<bool>().value_or(false);
 }
+
+int connect(std::string ipaddress, int port) {
+    struct sockaddr_in addr = {0};
+    addr.sin_addr.s_addr = INADDR_ANY;
+    if (!ipaddress.empty()) {
+        addr.sin_addr.s_addr = inet_addr(ipaddress.c_str());
+    }
+    addr.sin_family = AF_INET;
+    addr.sin_port = htons( port );
+
+    int sd = socket(AF_INET, SOCK_STREAM, 0);
+    if(connect(sd, (struct sockaddr *)&addr, sizeof(addr)) < 0) {
+        BOOST_LOG_TRIVIAL(warning) << "connect('" << ipaddress << "',"
+                                   << port << "') failed --> '"<< ::strerror(errno) << "'";
+        return -1;
+    }
+    return sd;
+}
+
+ExternalPhaseCoordinator::ExternalPhaseCoordinator(std::string host, int port) :
+      m_host(host), m_port(port), m_socket(connect(host, port)){
```

Comment:
Not to self, this is where the connect function is called and the ip connection is made.

Reply:
A comment would be good. It's clear, but I also read right past it and then went searching for it.

Reply:
done.